# Bike Sharing Demand 🚲 
# EDA

In [8]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/bike-sharing-demand/sampleSubmission.csv
/kaggle/input/bike-sharing-demand/train.csv
/kaggle/input/bike-sharing-demand/test.csv


In [3]:
data_path = '/kaggle/input/bike-sharing-demand/'

In [5]:
train = pd.read_csv(data_path + 'train.csv') # 훈련 데이터
test = pd.read_csv(data_path + 'test.csv') # 테스트 데이터

In [6]:
submission = pd.read_csv(data_path + 'sampleSubmission.csv') # 제출 샘플 데이터

In [7]:
# 훈련데이터와 테스트 데이터 크기를 확인

train.shape, test.shape

((10886, 12), (6493, 9))

- 훈련 데이터는 10,886행 **12열**로 구성
- 테스트 데이터는 6,493행 **9열**로 구성
- ❓ Note that: 피처의 개수가 서로 다르다
    - 어떤 피처 데이터를 담고있는지 살펴봅시다

In [9]:
train.head()

datetime  season  holiday  workingday  weather  temp   atemp  \
0  2011-01-01 00:00:00       1        0           0        1  9.84  14.395   
1  2011-01-01 01:00:00       1        0           0        1  9.02  13.635   
2  2011-01-01 02:00:00       1        0           0        1  9.02  13.635   
3  2011-01-01 03:00:00       1        0           0        1  9.84  14.395   
4  2011-01-01 04:00:00       1        0           0        1  9.84  14.395   

   humidity  windspeed  casual  registered  count  
0        81        0.0       3          13     16  
1        80        0.0       8          32     40  
2        80        0.0       5          27     32  
3        75        0.0       3          10     13  
4        75        0.0       0           1      1